### PR TITLE
Add functionality and tests from cmfplone

### PR DESCRIPTION
**Pull request by mauritsvanrees**
This also adds an uninstall profile.
And we no longer count on being installed by CMFPlone, especially in tests.
And this only uses proper `plone.app.testing`.
Maybe I should split this PR up, but the changes are intertwined (copying tests meant I had to refactor the test setup.)
This is in preparation of PLIP https://github.com/plone/Products.CMFPlone/issues/1775

This might work fine on Plone 4.3 and higher actually. But probably best to only do this for 5.2.  Or really wait for Plone 6, where this would then be merged together with some changes in CMFPlone (see [initial branch](https://github.com/plone/Products.CMFPlone/tree/remove-qi-completely-master)) and plone.app.upgrade that would remove QI from the core.
My idea is to keep the QI working outside of the core, for those few poor souls that somehow still need it even in Plone 6. At that time (or earlier) we should add a Travis config, so the QI can be tested outside of Jenkins.

### CODE REVIEW

File: Products/CMFQuickInstallerTool/QuickInstallerTool.py [hunk 1]
```diff
@@ -90,6 +118,7 @@ class QuickInstallerTool(UniqueObject, ObjectManager, SimpleItem):
 
     meta_type = 'CMF QuickInstaller Tool'
     id = 'portal_quickinstaller'
+    # toolicon = 'skins/plone_images/product_icon.png'
```
Comment (gforcada) [inline Products/CMFQuickInstallerTool/QuickInstallerTool.py]: @mauritsvanrees leftover? 🙂

File: Products/CMFQuickInstallerTool/QuickInstallerTool.py [hunk 2]
```diff
@@ -817,5 +846,94 @@ def getInstanceHome(self):
...
+                step = upgradestep['step']
+                step.doStep(setup)
+        version = str(profile['version'])
+        if version == 'latest':
+            version = self.getLatestUpgradeStep(profile_id)
+        setup.setLastVersionForProfile(profile_id, version)
+
 
 InitializeClass(QuickInstallerTool)
+# TODO: we probably want this, copied from CMPlone:
```
Comment (gforcada) [inline Products/CMFQuickInstallerTool/QuickInstallerTool.py]: @mauritsvanrees I usually tend to add TODOs as `TODO(gforcada): lorem ipsum...`. Question though, when would it be a good time to decide if we want to copy it or not? 🤔

File: Products/CMFQuickInstallerTool/interfaces/portal_quickinstaller.py [hunk 3]
```diff
@@ -22,7 +22,10 @@ def isProductInstallable(productname):
 
     def isProductAvailable(productname):
         ''' is the product directory present (to check if it has been deleted
-            from the Filesystem '''
+            from the Filesystem
+
+        Deprecated: use isProductInstallable instead.
```
Comment (gforcada) [inline Products/CMFQuickInstallerTool/interfaces/portal_quickinstaller.py]: Should we raise a warning here? https://docs.plone.org/develop/styleguide/deprecation.html

File: Products/CMFQuickInstallerTool/QuickInstallerTool.py (hunk 1 above)
Comment (mauritsvanrees) [reply Products/CMFQuickInstallerTool/QuickInstallerTool.py]: Removed now. I have copied the PNG icon from CMFPlone to `tool.png`.

File: Products/CMFQuickInstallerTool/QuickInstallerTool.py (hunk 2 above)
Comment (mauritsvanrees) [reply Products/CMFQuickInstallerTool/QuickInstallerTool.py]: Adding a name to a `TODO` comment is a good idea.

Actually, I see it is useless to call `registerToolInterface` here. When we register it, it should mean that a `getToolByName(context, 'portal_quickinstaller')` would try to get the `IQuickInstallerTool` utility first, and fall back to getting the tool. But even in current Plone, there is no such utility registered in zcml. I briefly tried to do that, but then Plone failed to start.
Even if the utility call *would* work, you would probably get a utility that does not really work: the QI *really* needs to be a tool, storing stuff in the database, especially `InstalledProducts`.

I have removed the commented out lines now.

File: Products/CMFQuickInstallerTool/interfaces/portal_quickinstaller.py (hunk 3 above)
Comment (mauritsvanrees) [reply Products/CMFQuickInstallerTool/interfaces/portal_quickinstaller.py]: This is only in an interface definition. The implementation of this method in the tool itself already gives a proper warning (even in Plone 4).

**Comment by mauritsvanrees** [comment]
I have cleaned up the TODO remarks and addressed your points from above.

The failing test on 4.3 is real (installing `CMFPlacefulWorkflow` fails because no profiles are found). I tried fixing it, but completely failed. One other test would need a small change to pass on Python 2.6. So let's not do this for 4.3.

Technically, we could easily include it on Plone 5.1, and 5.0 would be fine too, but we don't really gain anything by it in those versions. It's all a bit cleaner though, which can be nice.

I say we make a maintenance branch 1.x (well, *I* can make that) and use that on 4.3, 5.0 and 5.1. Use master branch on 5.2 and bump the version to 2.0.0. Not *really* breaking changes, but seems good as this branch basically prepares for end-of-life of this package.

**Comment by mauritsvanrees** [comment]
Okay, I have made a maintenance branch for Products.CMFQuickInstaller. I said 1.x above, but I meant 3.x of course.
I have updated coredev 4.3, 5.0 and 5.1 to use this branch.
I will now merge this PR so 5.2 uses it.

**Pull request merged**